docs/src/tutorials/symbmc.md

```diff
@@ -0,0 +1,655 @@
+The first limitation is the absence of a termination guarantee. More specifically, this algorithm terminates if and only if `R` is finite. For example:
+```tla
+VARIABLE 
+  \* @type: Int;
+  x
+
+Init == x = 0
+Next == x' = x + 1
+```
+Defines a states space, for which `R = N`, so the above algorithm never terminates.
+Further, it is, in the general case, difficult or impossible to compute `S0` or `Successor(s)`. 
+Consider the following specification:
+```tla
+VARIABLE x
+
+Successor(n) == IF n % 2 = 0 THEN n \div 2 ELSE 3*n + 1 
+
+RECURSIVE kIter(_,_)
+kIter(a,k) == IF k <= 0 THEN a ELSE Successor(kIter(a, k-1))
+
+ReachesOne(a) == \E n \in Nat: kIter(a,n) = 1
+
+Init == x \in { n \in Nat: ~ReachesOne(n) }
+```
+The specification encodes the [Collatz conjecture](...), so computing `S0` is equivalent to proving or disproving the conjecture, which remains an open problem at present. It is therefore unreasonable to expect any model checker to be able to accept such input, despite the fact that the condition is easily describable in first-order logic.
+A similar problem occurs in computing `Successor(s)`; the relation between variables `vi` and `vi'` may be given by means of an implicit function or uncomputable expression.
+
+Therefore, most solvers impose the following constraints, which make computing `S0` and `Successor(s)` possible without any sort of specialized solver:
+The specification must have the shape
+```tla
+VARIABLE v1,...,vk
+
+Init == /\ v1 \in F1()
+        /\ v2 \in F2(v1)
+        ...
+        /\ vk \in Fk(v1,...,v{k-1})
+
+Next == /\ CondN(v1,...,vk)
+        /\ v1' \in G1(v1,...,vk)
+        /\ v2' \in G2(v1,...,vk, v1')
+        ...
+        /\ vk' \in Gk(v1,...,vk, v1',...,v{k-1}')
+```
+or some equivalent form, in which variable values in a state can be iteratively computed, one at a time, by means of an explicit formula, which uses only variables computed so far.
+For instance,
+```tla
+VARIABLE x,y
+
+Init == /\ x \in 1..0
+        /\ y \in { k \in 1..10, k > x }
+
+Next == \/ /\ x > 5
+           /\ x' = x - 1
+           /\ y' = x' + 1
+        \/ /\ x <= 5
+           /\ y' = 5 - x
+           /\ x' = x + y'
+```
+
+allows the solver to compute both `S0` as well as `Successor(s)`, for any `s`, by traversing the conjunctions in the syntax-imposed order. 
+
+However, even in a situation where states are computable, and `R` is finite, the size of `R` itself might be an issue in practice. We can create specifications of arbitrary size:
+```tla
+VARIABLE v1,...,vk
+
+Init == /\ v1 = 0
+        ...
+        /\ vk = 0
+Next == \/ /\ v1' = (v1 + 1) % C
+           /\ UNCHANGED <<v2,...,vk>>
+        \/ /\ v2' = (v2 + 1) % C
+           /\ UNCHANGED <<v1,v3,...,vk>>
+        ...
+        \/ /\ vk' = (vk + 1) % C
+           /\ UNCHANGED <<v1,...,v{k-1}>>
+
+```
+will have `C^k` distinct states, despite its rather simplistic behavior.
+
+## Symbolic bounded model checking
```

Comment:
It's possible, but bounded explicit-state MC is not really interesting, it's just a limit on the depth in DFS/BFS

Reply:
I think it could help build up the conceptual pieces tho. The fact that it's not interesting on its own might even be something that could be used to help the reader see why adding the symbolic checking on top of the bounds is cool. It might also help give opportunity to then say why the symbolic checking is bounded.

Reply:
Yeah, it's a good idea. BFS is essentially implementing bounded model checking. The difference though that you know when you have converged in BFS.